components/brave_new_tab_ui/containers/newTab/gridTile.tsx

```diff
@@ -3,158 +3,151 @@
-    if (event.key === 'Escape') {
-      this.setState({ showMenu: false })
-    }
-  }
-
-  handleClickOutside = (event: Event) => {
-    if (
-      this.tileMenuRef &&
-      this.tileMenuRef.current &&
-      !this.tileMenuRef.current.contains(event.target)
-    ) {
-      this.setState({ showMenu: false })
-    }
-  }
+export function SiteTile (props: { site: NewTab.Site, isMenuShowing?: boolean, children?: React.ReactNode, draggable?: ReturnType<typeof useSortable> }) {
+  const { site, isMenuShowing, children, draggable } = props
+  const style = useMemo(() => ({
+    transform: CSS.Transform.toString(draggable?.transform || null),
+    transition: draggable?.transition,
+    opacity: draggable?.isDragging ? 0 : 1
+  }), [draggable])
+
+  return <Tile
+    ref={draggable?.setNodeRef} {...draggable?.attributes} {...draggable?.listeners}
+    isDragging={!!draggable?.isDragging}
+    isMenuShowing={!!isMenuShowing}
+    title={site.title}
+    href={site.url}
+    style={style}>
+    {children}
+    <TileFavicon src={generateGridSiteFavicon(site)} />
+    <TileTitle>{site.title}</TileTitle>
+  </Tile>
+}
 
-  handleMoveOutside = (event: Event) => {
-    if (
-      this.tileMenuRef &&
-      this.tileMenuRef.current &&
-      !this.tileMenuRef.current.contains(event.target)
-    ) {
-      this.setState({ showMenu: false })
+function TopSite (props: Props) {
+  const { siteData, disabled } = props
+
+  const tileMenuRef = useRef<any>()
+  const sortable = useSortable({ id: siteData.id, disabled })
+  const [showMenu, setShowMenu] = useState(false)
+
+  const handleClickOutside = useCallback((e: Event) => {
+    if (!tileMenuRef.current || tileMenuRef.current.contains(e.target)) { return }
+    setShowMenu(false)
+  }, [])
+
+  const handleMoveOutside = useCallback((e: Event) => {
+    if (!tileMenuRef.current || tileMenuRef.current.contains(e.target)) { return }
+    setShowMenu(false)
+  }, [])
+
+  const onKeyPressSettings = useCallback((e: KeyboardEvent) => {
+    if (e.key === 'Escape') { setShowMenu(false) }
+  }, [])
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside)
+    document.addEventListener('mouseMove', handleMoveOutside)
+    document.addEventListener('keydown', onKeyPressSettings)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+      document.removeEventListener('mousemove', handleMoveOutside)
+      document.removeEventListener('keydown', onKeyPressSettings)
     }
-  }
+  }, [handleClickOutside, handleMoveOutside, onKeyPressSettings])
 
-  onIgnoredTopSite (site: NewTab.Site, e: Event) {
+  const onShowTileMenu = useCallback((e: React.MouseEvent) => {
     e.preventDefault()
-    this.setState({ showMenu: false })
-    this.props.actions.tileRemoved(site.url)
-  }
+    setShowMenu(true)
+  }, [])
 
-  onEditTopSite (site: NewTab.Site, e: Event) {
+  const onIgnoredTopSite = useCallback((site: NewTab.Site, e: React.MouseEvent) => {
```

Comment:
Seems like there's a mix of naming here for handlers - some are handleXYZ and others are onXYZ. I personally like the handle prefix for names of functions sent to events, but it's subjective and we don't yet have a style guide for this, so I'll leave it up to you as long as it's consistent.

Reply:
Renamed to handle. I'm partial to the on prefix, but only because it's short :D